### PR TITLE
Deployment for resource groups

### PR DESCRIPTION
**Pull request by Maryonsin**
Please review

### CODE REVIEW

File: resources/resourcegroup/azuredeploy.jsonc [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    "additionalTags": {
+      "type": "object",
+      "defaultValue": {},
+      "metadata": {
+        "description": "Additional tags that must be applied to each resource. All resources will be tagget by default with environment and application name"
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "North Europe",
```
Comment (niktho) [inline resources/resourcegroup/azuredeploy.jsonc:33]: @jlyshoel , I see that the arm-ttk is telling to use the  resourceGroup().location, but I assume that isn't possible when you are creating a resource group? One other way is to remove the default value for this parameter.

File: resources/resourcegroup/azuredeploy.jsonc [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    },
+    "principalId": {
+      "type": "string",
+      "metadata": {
+        "description": "principalId if the user that will be given contributor access to the resourceGroup"
+      }
+    },
+    "roleDefinitionId": {
+      "type": "string",
+      "defaultValue": "b24988ac-6180-42a0-ab88-20f7382dd24c",
```
Comment (niktho) [inline resources/resourcegroup/azuredeploy.jsonc:46]: I would assume that this defaultValue should not be in the open source repository. Meaning that we should create a deployment template in the IOC github repository calling this template with the correct parameter/default value.

**Review by niktho — COMMENTED** [review]
Can we rename the folder **resourcegroup** to **resourceResourceGroup** so it's following the same pattern as the other folders?

File: resources/resourcegroup/azuredeploy.jsonc (hunk 1 above)
Comment (jlyshoel) [reply resources/resourcegroup/azuredeploy.jsonc:33]: Hm, good question. resourceGroup().location is probably not possible when creating the resource group in question. If the ttk is recommending we remove the default, I would tend to agree and say we assign it from a parameters file or directly on CLI

File: resources/resourcegroup/azuredeploy.jsonc (hunk 2 above)
Comment (jlyshoel) [reply resources/resourcegroup/azuredeploy.jsonc:46]: I would normally agree, but in this case the ID seems to be an Azure constant [1] - not something internal. I vote for keeping this default, as the description states "default is contributor".

[1] https://docs.microsoft.com/en-us/azure/role-based-access-control/role-definitions#role-definition-example

Comment (niktho) [reply resources/resourcegroup/azuredeploy.jsonc:46]: Ah, didn't know. Thanks for researching.

File: resources/resourcegroup/azuredeploy.jsonc (hunk 1 above)
Comment (Maryonsin) [reply resources/resourcegroup/azuredeploy.jsonc]: Changed to "defaultValue":  {}, but check still failing

Comment (lp-code) [reply resources/resourcegroup/azuredeploy.jsonc]: That's not the same as removing the default, and it's not a string either. :-)

Comment (niktho) [reply resources/resourcegroup/azuredeploy.jsonc]: @lp-code , Maryon has fixed it now :)

**Comment by lp-code** [comment]
A general remark: I don't think we should put this template under "resources", since these are meant to be "stuff that can go into a resource group". This template will have to be deployed at subscription level, which is a different class of things. We could add a new directory for this kind of templates.

**Pull request merged**